docs/source/eventproc.rst

```diff
@@ -0,0 +1,201 @@
+    >>> conn = Connection("api.example.com:4411", "C27489F2BFACE794A3")
+
+
+However, the `Connection` object will also look for the `BTRDB_ENDPOINTS` and `BTRDB_API_KEY` environment variables if no arguments are supplied.
+
+.. code-block:: python
+
+    >>> from btrdbextras import Connection
+    >>> conn = Connection()
+
+
+Interactions
+------------
+
+The following calls can be made to platform to manage your event handling code.
+
+Listing Known Hooks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can think of hooks as just labels that represent a specific point during the execution of some process (such as importing data).  In order to see the allowed hooks, you may invoke the `hooks` function.  Over time, new `hooks` will be added to the platform allowing you to integrate your own code into different processes.
+
+.. code-block:: python
+
+    >>> from btrdbextras import Connection
+    >>> from btrdbextras.eventproc import hooks
+
+    >>> conn = Connection()
+    >>> hooks(conn)
+    ['ctingress.on_complete']
+
+
+Registering Event Handlers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The event handlers are supplied by you.  This is code you would like executed when a specific hook/event has occurred within the larger platform. Your Python callable, typically just a function, will be pickled using the `dill` library for execution in the platform.  As such there are certain Python code serialization topics to keep in mind such as: total size of the callable, libraries available in the execution environment, etc.
+
+To submit a new event handler, you can use the `register` decorator around your own Python callable.  Like all objects in this library, you can use the `help` function to view the docstring.
+
+.. code-block:: python
+
+    >>> from btrdbextras.eventproc import register
+    >>> help(register)
+    Help on function register in module btrdbextras.eventproc.eventproc:
+
+    register(conn, name, hook, notify_on_success, notify_on_failure, dependencies, flags=None)
+        decorator to submit (register) an event handler function
+
+        Parameters
+        ----------
+        conn: Connection
+            btrdbextras Connection object containing a valid address and api key.
+        name: str
+            Friendly name of this event handler for display purposes.
+        hook: str
+            Name of the hook that this event handler responds to.
+        notify_on_success: str
+            Email address of user to notify when event handler completes successfully.
+        notify_on_failure: str
+            Email address of user to notify when event handler does not complete
+            successfully.
+        dependencies: str
+            A string containing required libraries and versions.  Format should follow
+            a standard `pip freeze` or requirements.txt file.  The process running your
+            handler will `pip install -r` this content.
+        flags: list of str
```

Comment:
Let's make the empty list behavior explicit here. "An empty list will match all flags."

Reply:
done!